spec/routes/api/vm_spec.rb

```diff
@@ -43,55 +92,193 @@
       end
 
-      it "success" do
+      it "success single" do
+        get "/api/project/#{project.ubid}/location/#{vm.location}/vm"
+
+        expect(last_response.status).to eq(200)
+        expect(JSON.parse(last_response.body)["values"].length).to eq(1)
+        expect(JSON.parse(last_response.body)["next_cursor"]).to be_nil
+        expect(JSON.parse(last_response.body)["count"]).to eq(1)
+      end
+
+      it "success multiple" do
+        Prog::Vm::Nexus.assemble("dummy-public-key", project.id, name: "dummy-vm-2")
+
         get "/api/project/#{project.ubid}/location/#{vm.location}/vm"
 
         expect(last_response.status).to eq(200)
-        expect(JSON.parse(last_response.body).length).to eq(1)
+        expect(JSON.parse(last_response.body)["values"].length).to eq(2)
+        expect(JSON.parse(last_response.body)["next_cursor"]).to be_nil
+        expect(JSON.parse(last_response.body)["count"]).to eq(2)
+      end
+
+      it "success order column" do
+        Prog::Vm::Nexus.assemble("dummy-public-key", project.id, name: "dummy-vm-2")
+
+        get "/api/project/#{project.ubid}/location/#{vm.location}/vm", {
+          order_column: "name"
+        }
+
+        expect(last_response.status).to eq(200)
+        expect(JSON.parse(last_response.body)["values"][0]["id"]).to eq(vm.ubid)
+      end
+
+      it "success page size 1" do
+        Prog::Vm::Nexus.assemble("dummy-public-key", project.id, name: "dummy-vm-2")
+
+        get "/api/project/#{project.ubid}/location/#{vm.location}/vm", {
+          page_size: 1
+        }
+
+        expect(last_response.status).to eq(200)
+        expect(JSON.parse(last_response.body)["values"].length).to eq(1)
+        expect(JSON.parse(last_response.body)["count"]).to eq(2)
+      end
+
+      it "success page size 2" do
+        Prog::Vm::Nexus.assemble("dummy-public-key", project.id, name: "dummy-vm-2")
+
+        get "/api/project/#{project.ubid}/location/#{vm.location}/vm", {
+          page_size: 2
+        }
+
+        expect(last_response.status).to eq(200)
+        expect(JSON.parse(last_response.body)["values"].length).to eq(2)
+      end
+
+      it "success page size less than count" do
+        vm_2 = Prog::Vm::Nexus.assemble("dummy-public-key", project.id, location: vm.location, name: "dummy-vm-2")
+
+        get "/api/project/#{project.ubid}/location/#{vm.location}/vm", {
+          page_size: 1,
+          order_column: "name"
+        }
+
+        expect(JSON.parse(last_response.body)["next_cursor"]).to eq(vm_2.ubid)
+        expect(last_response.status).to eq(200)
+        expect(JSON.parse(last_response.body)["values"].length).to eq(1)
+      end
+
+      it "success negative page size" do
+        get "/api/project/#{project.ubid}/location/#{vm.location}/vm", {
+          page_size: -1
+        }
+
+        expect(last_response.status).to eq(200)
+        expect(JSON.parse(last_response.body)["values"].length).to eq(1)
+      end
+
+      it "success non numeric page size" do
```

Comment:
IMHO, this shouldn't be a test in the vm spec. We should test this generic pagination feature in the pagination module.

Reply:
`paginated_result` has the branching, not the vm listing route